src/NonEmptyArray2v.ts

```diff
@@ -0,0 +1,316 @@
+/**
+ * @file Data structure which represents non-empty arrays
+ */
+import { Monad1 } from './Monad'
+import * as A from './Array'
+import { Comonad1 } from './Comonad'
+import { FunctorWithIndex1 } from './FunctorWithIndex'
+import { TraversableWithIndex1, TraverseWithIndex1 } from './TraversableWithIndex'
+import { FoldableWithIndex1 } from './FoldableWithIndex'
+import { Ord } from './Ord'
+import { fold, getMeetSemigroup, getJoinSemigroup, Semigroup } from './Semigroup'
+import { Option, some, none } from './Option'
+import { Setoid } from './Setoid'
+import { compose, Predicate, Refinement } from './function'
+import { Traverse1 } from './Traversable'
+import { Sequence1 } from './Traversable2v'
+
+declare module './HKT' {
+  interface URI2HKT<A> {
+    NonEmptyArray2v: NonEmptyArray<A>
+  }
+}
+
+export const URI = 'NonEmptyArray2v'
+
+export type URI = typeof URI
+
+/**
+ * @since 1.14.5
+ */
+export interface NonEmptyArray<A> extends Array<A> {
+  0: A
+  map: <B>(f: (a: A, index: number, nea: NonEmptyArray<A>) => B) => NonEmptyArray<B>
+  concat: (as: Array<A>) => NonEmptyArray<A>
+  reverse: () => NonEmptyArray<A>
+}
+
+/**
+ * @since 1.14.5
+ */
+export function make<A>(head: A, tail: Array<A>): NonEmptyArray<A> {
+  return [head, ...tail] as any
```

Comment:
I believe this can be `return <NonEmptyArray<A>>[head, ...tail]`.

Reply:
@yawaramin thanks for reviewing

```
<NonEmptyArray<A>>[head, ...tail]
```

raises the following error

```
Property '0' is missing in type 'A[]' but required in type 'NonEmptyArray<A>'
```

I'm afraid that casting to `any` is the only way to please TypeScript here, is not a big deal though